sdk/go/common/util/cmdutil/child_windows.go

```diff
@@ -20,11 +20,28 @@ package cmdutil
 import (
 	"os"
 	"os/exec"
+	"syscall"
 
 	multierror "github.com/hashicorp/go-multierror"
 	ps "github.com/mitchellh/go-ps"
 )
 
+// killProcessGroup kills a process group by calling Process.Kill()
+// (which calls TerminateProcess on Windows) on all processes in the group.
+//
+// This is different from [KillChildren] which only kills child processes.
+//
+// This is a helper function for TerminateProcessGroup;
+// a Unix version with the same signature exists in child_unix.go.
+func killProcessGroup(proc *os.Process) error {
+	if err := KillChildren(proc.Pid); err != nil {
+		return err
+	}
+
+	// Kill the root process since KillChildren only kills child processes.
```

Comment:
Why does windows kill the root process but unix doesn't?

Reply:
It does. The killProcessGroup for Unix calls KillChildren, which sends the signal to the entire group, which includes the root. 
(Signal to -pid goes to the entire group.)

We have to do it manually on Windows because the TerminateProcess syscall (which is what os.Process.Kill uses) doesn't go to the whole group. The implementation of KillChildren for Windows only killed the child processes. So there was a discrepancy in Windows vs Unix behaviors of our KillChildren implementations, which is why killProcessGroup has to also go and kill the root process.

Reply:
So KillChildren(pid) on windows is just kill the children, but KillChildren on unix is kill children and self?

That's upsetting.

Reply:
Yeah. It hasn't been a problem because the only place the function is used, the root process is killed afterwards anyway.

Reply:
Can we just remove the KillChildren function inline it's parts into KillProcessGroup and just have that?

Reply:
I didn't want to break it in case someone was using it directly. Next PR deprecates it, though.

I'm open to deleting it if we're not concerned about breaking it but it probably makes sense to do that in the same PR where we stop using it.

Reply:
I'm not concerned, sdk/go/common isn't exactly a stable public api and most of it is only used by pkg.

But if the next PR removes the use of it then I'm fine with this change as is, and we'll remove it in that one.